### DISCUSSION TITLE
Custom changes in VannaFlaskAPI

### DISCUSSION THREAD

**Discussion by zawalich** [vanna-ai/vanna#671] [Q&A]
Hello @zainhoda , I would like to make changes in VannFlaskAPI functions declared [here](https://github.com/vanna-ai/vanna/blob/main/src/vanna/flask/__init__.py#L199) - e.g. add some keys to returned JSON dictionaries. How can I achieve that? Pre- and post-filtering with `super` (shown [here](https://www.thedigitalcatonline.com/blog/2014/05/19/method-overriding-in-python/)) doesn't work, as these methods are declared directly in `__init__`. It would be an ideal solution, as it avoids maintaining source code, only the customizations.

**Comment by zawalich** [comment]
I have implemented a workaround for this - in case you need it, let me know, and I will describe it here.

**Reply by zainhoda** [reply]
I'd be open to a PR if you've made a modification that allows easily adding functions

**Reply by zawalich** [reply]
Unfortunately, my solution is of a "hack" one rather than an elegant  - I am overwriting functions with my definitions in Flask in-between of serving.
